app/components/editor/problem-solve.tsx

```diff
@@ -0,0 +1,184 @@
+    const {solverMetadata, setSolverMetadata} = props;
+    const emptySolverMetadata = {
+        type: "FORMULA",
+        variants: [],
+        correct: null,
+        formula: null
+    };
+    
+    const setType = (type: any) => {
+        const _solverMetadata = {...emptySolverMetadata, type};
+
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const handleFormulaChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const _solverMetadata = {...solverMetadata, formula: e.target.value};
+        
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const handleNumberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const input = e.target.value;
+        const _solverMetadata = {
+            ...solverMetadata,
+            correct: {
+                type: "NUMBER", 
+                number: input ? parseInt(e.target.value) : null, 
+                string: null, 
+                index: null
+            }
+        };
+        
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const handleVariantChange = (e: React.ChangeEvent<HTMLInputElement>, index: number) => {
+        let _solverMetadata = {...solverMetadata};
+        let variantToChange = solverMetadata.variants[index];
+        variantToChange.string = e.target.value;
+        _solverMetadata.variants[index] = variantToChange;
+
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const setCorrectVariant = (variant: any) => {
+        let _solverMetadata = {...solverMetadata};
+        _solverMetadata.correct.index = parseInt(variant);
+        
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const addVariant = () => {
+        let _solverMetadata = {...solverMetadata};
+        if(_solverMetadata.variants.length === 0) {
+            _solverMetadata.correct = {
+                type: "INDEX",
+                number: null,
+                string: null,
+                index: 0
+            }
+        }
+
+        _solverMetadata.variants.push({
+            type: "STRING",
+            number: null,
+            string: "",
+            index: null
+        });
+        
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const removeVariant = (index: number) => {
+        let _solverMetadata = {...solverMetadata};
+        _solverMetadata.variants.splice(index);
+        if(_solverMetadata.variants.length === 0) {
+            _solverMetadata.correct = null;
+        } else {
+            _solverMetadata.correct = {
+                type: "INDEX",
+                number: null,
+                string: null,
+                index: 0
+            }
+        }
+        
+        setSolverMetadata(_solverMetadata);
+    }
+
+    const solveContent = () => {
+        if(solverMetadata.type === 'FORMULA') {
+            return (
+                <div>
+                    <ReactKatex breakLine={true} strict={false} children={solverMetadata.formula}></ReactKatex>
```

Comment:
Preview specifically like a formula, not a text

```suggestion
                    <ReactKatex breakLine={true} strict={false} children={`$${solverMetadata.formula}$`}></ReactKatex>
```